src/kv/kv.h

```diff
@@ -1776,13 +1827,21 @@ namespace kv
 
           if (h)
           {
-            h->add_result(reqid, version, data_.data(), data_.size());
+            if (c->type() == ConsensusType::RAFT)
```

Comment:
Pushing knowledge of the consensus in the KV is extremely unfortunate. There should instead be a variant of history, and nodes using BFT should instantiate that instead.